app/handlers.go

```diff
@@ -292,3 +348,32 @@ func printObj(w http.ResponseWriter, v interface{}) {
 	vBytes, _ := json.Marshal(v)
 	fmt.Fprint(w, string(vBytes))
 }
+
+func extractChangedAlbumID(resp db.WriteResponse, r *http.Request) error {
+	if (len(resp.Changes) > 0){
+		newAlbumID := getAlbumIDFromReturnValues(resp.Changes[0].NewValue)
+		oldAlbumID := getAlbumIDFromReturnValues(resp.Changes[0].OldValue)
+
+		var err error
+		if newAlbumID != oldAlbumID {
+			err = SetAlbumPublic(newAlbumID, true, r)
+			if err != nil {
+				return err
+			}
+			err = SetAlbumPublic(oldAlbumID, false, r)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func getAlbumIDFromReturnValues(data interface{}) int {
+	m := data.(map[string]interface{})
+	var albumID int
+	if albumFloat, ok := m["album_id"].(float64); ok {
+		albumID = int(albumFloat)
```

Comment:
It looks like the albumID is being coerced to an int from a float64, will there be any truncation issues? uint is probably more appropriate.

Reply:
The value is an int in every other context including the DB, so it will always be an int coming through here. For whatever reason, the DB driver always returns a float64 and it extract it, you _have_ to cast it to an float64 before you can do anything with it.